src/objects/zif_abapgit_object.intf.abap

```diff
@@ -2,34 +2,36 @@ INTERFACE zif_abapgit_object
   PUBLIC .
 
   TYPES:
-    ty_deserialization_step TYPE string.
+    ty_deserialization_step TYPE string .
   TYPES:
-    ty_deserialization_step_tt TYPE STANDARD TABLE OF ty_deserialization_step
-                                          WITH DEFAULT KEY .
+    ty_deserialization_step_tt TYPE STANDARD TABLE OF ty_deserialization_step WITH DEFAULT KEY .
+  TYPES:
+    ty_complete_status TYPE c LENGTH 1 .
 
   DATA mo_files TYPE REF TO zcl_abapgit_objects_files .
-
   CONSTANTS:
     BEGIN OF gc_step_id,
       abap TYPE zif_abapgit_object=>ty_deserialization_step VALUE `ABAP`,
       ddic TYPE zif_abapgit_object=>ty_deserialization_step VALUE `DDIC`,
       late TYPE zif_abapgit_object=>ty_deserialization_step VALUE `LATE`,
-    END OF gc_step_id.
-
+    END OF gc_step_id .
   CONSTANTS c_abap_version_sap_cp TYPE progdir-uccheck VALUE '5' ##NO_TEXT.
   CONSTANTS c_abap_version_default TYPE progdir-uccheck VALUE 'X' ##NO_TEXT.
 
   METHODS serialize
     IMPORTING
       !io_xml TYPE REF TO zcl_abapgit_xml_output
+      !ii_log TYPE REF TO zif_abapgit_log
     RAISING
       zcx_abapgit_exception .
   METHODS deserialize
     IMPORTING
-      !iv_package TYPE devclass
-      !io_xml     TYPE REF TO zcl_abapgit_xml_input
-      !iv_step    TYPE ty_deserialization_step
-      !ii_log     TYPE REF TO zif_abapgit_log
+      !iv_package               TYPE devclass
+      !io_xml                   TYPE REF TO zcl_abapgit_xml_input
+      !iv_step                  TYPE ty_deserialization_step
+      !ii_log                   TYPE REF TO zif_abapgit_log
+    RETURNING
+      VALUE(rv_complete_status) TYPE ty_complete_status
```

Comment:
this is a central change to overall logic, please make sure to include it in the description of the pull request

Reply:
done